attorch/train.py

```diff
@@ -31,17 +31,25 @@ def early_stopping(model, objective, interval=5, patience=20, start=0, max_iter=
         max_iter:  maximum number of iterations before the iterator terminated
         maximize:  whether the objective is maximized of minimized
     """
+    training_status = model.training
+
+    def _objective(mod):
```

Comment:
Hm, that's a rather weird case. Bottom line is that we need the model to be in the evaluation mode (`train(mode=False)`) when performing evaluation on the validation set. I'd prefer to do the switch to the evaluation mode inside the early stopping code to ensure that the user doesn't have to remember doing this switch (as it was forgotten previously). 

In general the model should have two states : `train` vs `eval` and it is really up to the model implementer to override the necessary methods (e.g. `train` and `eval`) if they don't want this status flip to propagate all the way to all submodules.